healthcheck/tools/db/ssl.go

```diff
@@ -52,103 +48,53 @@ func LastSSLError() error {
 	return lastSSLErr
 }
 
-func (cnf *Config) configureSSLDialInfo() error {
+func (cnf *Config) configureTLS() error {
 	config := &tls.Config{
 		InsecureSkipVerify: cnf.SSL.Insecure,
 	}
 
-	if len(cnf.SSL.PEMKeyFile) > 0 && len(cnf.SSL.CAFile) > 0 {
-		pemOk, err := isFileExists(cnf.SSL.PEMKeyFile)
-		if err != nil {
-			return fmt.Errorf("Failed to check if file with name %s exists, err: %v", cnf.SSL.PEMKeyFile, err)
-		}
-
-		caOk, err := isFileExists(cnf.SSL.CAFile)
-		if err != nil {
-			return fmt.Errorf("Failed to check if file with name %s exists, err: %v", cnf.SSL.CAFile, err)
-		}
-
-		if pemOk && caOk {
-			log.Debugf("Loading SSL/TLS PEM certificate: %s", cnf.SSL.PEMKeyFile)
-
-			certificates, err := tls.LoadX509KeyPair(cnf.SSL.PEMKeyFile, cnf.SSL.PEMKeyFile)
-			if err != nil {
-				return fmt.Errorf(
-					"Cannot load key pair from '%s' to connect to server '%s'. Got: %v",
-					cnf.SSL.PEMKeyFile,
-					cnf.DialInfo.Addrs,
-					err,
-				)
-			}
-
-			config.Certificates = []tls.Certificate{certificates}
-
-			log.Debugf("Loading SSL/TLS Certificate Authority: %s", cnf.SSL.CAFile)
-			ca, err := cnf.SSL.loadCaCertificate()
-			if err != nil {
-				return fmt.Errorf("Couldn't load client CAs from %s. Got: %s", cnf.SSL.CAFile, err)
-			}
-
-			config.RootCAs = ca
-
-			cnf.DialInfo.DialServer = tlsDial(config)
-		} else {
-			cnf.SSL = nil
-		}
+	if len(cnf.SSL.PEMKeyFile) == 0 || len(cnf.SSL.CAFile) == 0 {
+		return nil
 	}
 
-	return nil
-}
-
-func tlsDial(config *tls.Config) func(addr *mgo.ServerAddr) (net.Conn, error) {
-	return func(addr *mgo.ServerAddr) (net.Conn, error) {
-		conn, err := tls.Dial("tcp", addr.String(), config)
-		if err != nil {
-			log.Errorf("could not connect to %v. got: %v", addr, err)
-			lastSSLErr = err
-			return nil, err
-		}
-
-		if !config.InsecureSkipVerify {
-			dnsName := strings.SplitN(addr.String(), ":", 2)[0]
-			err = validateConnection(conn, config, dnsName)
-			if err != nil {
-				log.Errorf("Could not disable hostname validation. Got: %v", err)
-			}
-		}
-
-		return conn, err
+	pemOk, err := isFileExists(cnf.SSL.PEMKeyFile)
+	if err != nil {
+		return fmt.Errorf("Failed to check if file with name %s exists, err: %v", cnf.SSL.PEMKeyFile, err)
```

Comment:
I think it's better to use `errors` pkg here and below

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/757/commits/b7032b2da10a4b4eb30909772ea64bb56beb9f33